src/pal/src/misc/environ.cpp

```diff
@@ -634,11 +661,422 @@ SetEnvironmentVariableA(
             goto done;
         }
     }
+
     bRet = TRUE;
+
 done:
-    LOGEXIT("SetEnvironmentVariableA returning BOOL %d\n", bRet);    
+    LOGEXIT("SetEnvironmentVariableA returning BOOL %d\n", bRet);
     PERF_EXIT(SetEnvironmentVariableA);
     return bRet;
 }
 
+/*++
+Function:
+  ResizeEnvironment
+
+Resizes the PAL environment buffer.
+
+Parameters
+
+    newSize 
+           [in] New size of palEnvironment
+
+Return Values
+
+    TRUE on success, FALSE otherwise
+
+--*/
+BOOL ResizeEnvironment(int newSize)
+{
+    CPalThread * pthrCurrent = InternalGetCurrentThread();
+    InternalEnterCriticalSection(pthrCurrent, &gcsEnvironment);
+
+    if (newSize < palEnvironmentCount)
+    {
+        ASSERT("ResizeEnvironment: New size < current count!\n");
+        return FALSE;
+    }
+
+    palEnvironmentCapacity = newSize;
+
+    // If palEnvironment is null, realloc acts like malloc.
+    palEnvironment = (char**)realloc(palEnvironment, palEnvironmentCapacity * sizeof(char *));
```

Comment:
If realloc returns NULL, the original block of memory remains valid and unchanged. So this code will just drop it.